### PR TITLE
#ISC #21 HTTPS access endpoint

### PR DESCRIPTION
**Pull request by jdrew-sans**

### CODE REVIEW

File: srv/isc-agent/plugins/tcp/http/main.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import re
 
 from twisted.web import server, resource
-from twisted.internet import reactor, endpoints
+from twisted.internet import reactor, endpoints, ssl
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/main.py:6]: Can you rearrange this so the imports are in alphabetical order

File: srv/isc-agent/plugins/tcp/http/main.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 }
 logger = logging.getLogger(__name__)
 
+sslContext = ssl.DefaultOpenSSLContextFactory(settings.PRIVATE_KEY, settings.CERT_KEY)
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/main.py:21]: Python variables should be in snake case:
https://betterprogramming.pub/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841

File: srv/isc-agent/plugins/tcp/http/main.py [hunk 3]
```diff
@@ -42,6 +44,15 @@ def get_signature_score(rules, attributes):
     return score
 
 
+def http_endpoint(ports):
+    for port in ports:
+        endpoints.serverFromString(reactor, f'tcp:{port}').listen(server.Site(HTTP()))
+
+
+def https_endpoint():
+    endpoints.SSL4ServerEndpoint(reactor, 8001, sslContext).listen(server.Site(HTTP()))
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/main.py:53]: Port shouldn't be hardcoded

File: srv/isc-agent/plugins/tcp/http/main.py [hunk 4]
```diff
@@ -81,5 +92,5 @@ def render(self, request: Request):
 def handler(**kwargs):
     prepare_database()
     ports = kwargs.get('ports', default_ports)
-    for port in ports:
-        endpoints.serverFromString(reactor, f'tcp:{port}').listen(server.Site(HTTP()))
+    reactor.callInThread(http_endpoint(ports))
+    reactor.callInThread(https_endpoint)
```
Comment (bmyrtil-sans) [inline srv/isc-agent/plugins/tcp/http/main.py:96]: Why is one being called with ports and the other not being called at all? Also how does the callInThread work here?

File: srv/isc-agent/settings.py [hunk 5]
```diff
@@ -57,6 +57,10 @@
 )
 DATABASE_SESSION = Session(DATABASE_ENGINE)
 
+# CSS certification key and certificate
```
Comment (bmyrtil-sans) [inline srv/isc-agent/settings.py:60]: I dont think CSS is the write term here. CSS is more for frontend design stuff.

File: srv/isc-agent/settings.py [hunk 6]
```diff
@@ -57,6 +57,10 @@
...
+PRIVATE_KEY = os.path.join('key.pem')
+CERT_KEY = os.path.join('cert.pem')
+
 # PLUGINS
 # Eventually this value will be inferred from a settings file of some sort
 PLUGINS = [
```
Comment (bmyrtil-sans) [inline srv/isc-agent/settings.py:66]: Somewhere in the configs it needs to contain what ports should be used for HTTPS and what ports should be used for HTTP

File: srv/isc-agent/plugins/tcp/http/main.py (hunk 2 above)
Comment (bmyrtil-sans) [reply srv/isc-agent/plugins/tcp/http/main.py:21]: We also need to do a check that if there is no file at the location then we dont allow HTTPS to be used.

File: etc/CA/certs/04.pem [hunk 7]
```diff
@@ -0,0 +1,99 @@
+Certificate:
```
Comment (bmyrtil-sans) [inline etc/CA/certs/04.pem:1]: No certs or keys should be included in the repo. Its like exposing passwords. These certs we will expect to exist already on the Raspberry Pi so we only need to reference them by location we dont have to include them.

**Review by bmyrtil-sans — CHANGES_REQUESTED** [review]

File: etc/CA/certs/04.pem (hunk 7 above)
Comment (jdrew-sans) [reply etc/CA/certs/04.pem:1]: Understood

File: srv/isc-agent/plugins/tcp/http/main.py (hunk 1 above)
Comment (jdrew-sans) [reply srv/isc-agent/plugins/tcp/http/main.py:6]: Sure thing.

File: srv/isc-agent/settings.py (hunk 5 above)
Comment (jdrew-sans) [reply srv/isc-agent/settings.py:60]: Rookie mistake.  My bad.

**Review by bmyrtil-sans — APPROVED** [review]

**Pull request merged**